Ajuste na progress bar e uso de viewBinding e extensões do kotlin

KeysiJones opened the pull request:
- Ajuste na progressBar para desaparecer quando os dados forem carregados.
- Adicionada viewBinding e extensões do Kotlin para facilitar o acesso aos elementos das views sem precisar usar o findViewById.

Commit:
- Ajustada a progress bar para desaparecer quando os dados forem carregados.
- Adicionada viewBinding e extensões do Kotlin para facilitar o acesso aos elementos das views sem precisar usar o findViewById.

GuilhermeMBisotto commented on app/src/main/java/com/kdev/consultacep/overview/DadosFragment.kt:
```diff
@@ -17,17 +17,23 @@
 import com.kdev.consultacep.R
 import com.kdev.consultacep.databinding.FragmentDadosBinding
+import com.kdev.consultacep.model.Endereco
+import kotlinx.android.synthetic.main.fragment_dados.*
```
tu tá usando DataBinding/ViewBinding, então esses imposts de synthetics não deveriam existir

GuilhermeMBisotto commented on app/src/main/java/com/kdev/consultacep/overview/DadosFragment.kt:
```diff
@@ -59,7 +67,16 @@ class DadosFragment : Fragment() {
 
-        binding.viewModel = viewModel
+        viewModel.property.observe(viewLifecycleOwner, Observer<Endereco> { endereco ->
+            cepText.text = endereco.cep
```
essa sintaxe tu tá invocando os atributos do synthetic, deveria usar a sintaxe com o Binding, ex:

`viewModel.property.observe(viewLifecycleOwner, Observer<Endereco> { endereco ->
   binding.cepText.text = endereço-me.cep
   }`
   
   
   caso queria simplificar um pouco pode usar até um Binding.apply{}
   pra poder usar essa sintaxe de cepText.text

GuilhermeMBisotto commented on app/src/main/java/com/kdev/consultacep/overview/DadosViewModel.kt:
```diff
@@ -52,6 +52,11 @@ class DadosViewModel : ViewModel() {
+
+    }
+
+    fun getCep() : String? {
```
qual a necessidade desse metodo?

KeysiJones replied:
Esse método foi a tentativa de trazer os dados antes de eu conhecer o observer. A minha idéia era fazer um getter para cada atributo. Esqueci de remover hahahaha agora vou tirar ele.

GuilhermeMBisotto replied:
tendi, mas não tem muita necessidade mesmo de tu ter gentes para cada atributo, seria apenas poluição da classe, mas caso tu queira enviar os dados "mais mastigados" poderia usar uma sealed class:

`sealed class Presentation {
    object Loading : Presentation()
    data class Error(val message: String) : Presentation()
    data class Success(val cep: String, val ...) : Presentation()
}`

com isso tu pode criar um liveData do tipo da sealed class e no Fragment pode resolver via when

`when (presentation.value) {
        Presentation.Loading -> {
        }
        is Presentation.Error -> {
        }
        is Presentation.Success -> {
        }
    }`

Commit:
- Remoção do Synthetics e de método não usado.
- Ajuste para usar o viewBinding em vez do plugin synthetics para ter null safety

Commit:
remoção de variáveis e código não usado.

GuilhermeMBisotto reviewed (approved):
okay
